### ISSUE TITLE
REQ: keys.shared.js and sync.user.js to allow shared keys shared amongst many users

### ISSUE THREAD

**Issue by triune** [jonatkins/ingress-intel-total-conversion#1018]
If the sync data is stored in the IITC-SYNC-DATA-V2 folder in Google Drive, would it be possible to instead specify a different folder to store key data? Perhaps, say in KEYS-SHARED folder? The sync and shared plugins would most likely both need to be updated to allow the viewing of keys added to this common repository accessible by multiple users. I envision that a Google Drive folder could be shared with others (access managed from within drive) and all users would be able to contribute keys to portals that the entire group could view. Each user would only be able to add/delete their own keys but see all key totals from the group for each portal. This would make managing large field ops among friends much easier!

**Comment by SneakyWhoami** [comment]
Or if the bookmarks plugin coild see the key counts on import/export, would that do it?

**Comment by nexushoratio** [comment]
Syncing with one person mostly works because you have to go out of your way to modify information in multiple tabs at the same time.   But, if you work at it, and watch carefully, say two windows open right next to each other, and star portals in one window, then immediately star one in another, you can see one disappear due to the race conditions.

Current implementations of syncing work on the collection level (bookmarks, keys, etc), not on the item level within any given collection.

Opening up sharing to multiple people, who might actually modify the same collection at the same time, it would likely increase the chances of losing data.  And you're less likely to notice it than if you do it to yourself.

In short, a lot of work would have to be done to improve internals before even attempting this.

**Comment by triune** [comment]
Since every drive user has a unique id and every portal a unique id, when a key count is incremented for a particular portal could the value saved be a concatenation of the two values? When saved to drive this would avoid the multiple user race condition (not single user which should be a low priority to fix anyway). When reading values from drive sync your going to want to see total key counts per portal as well as individual player key counts for a portal. However, I'm not sure there is a mechanism to save user name / game name / drive name to the sync data that can be bubbled up to the interface for displaying who has the keys. Might have to tie game name to drive user unique id?
